packages/opentelemetry-context-base/src/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Default type for functions */
+// tslint:disable:no-any
+export type Func<T> = (...args: any[]) => T;
+
+/** A key/value object for anything that can be stored in the context */
+export interface ContextStore {
+  [key: string]: unknown;
+}
+
+export interface ContextManager {
+  /**
+   * Binds the trace context to the given function.
+   * This is necessary in order to create child spans correctly in functions
+   * that are called asynchronously (for example, in a network response
+   * handler).
+   * @param fn A function to which to bind the trace context.
+   */
+  wrapFunction<T>(fn: Func<T>): Func<T>;
+
+  /**
+   * Binds the trace context to the given event emitter.
+   * This is necessary in order to create child spans correctly in event
+   * handlers.
+   * @param emitter An event emitter whose handlers should have
+   *     the trace context binded to them.
+   */
+  wrapEmitter(emitter: NodeJS.EventEmitter): void;
+
+  /**
+   * Get the current key/value storage from the current context.
+   * There are specific case where the context is not propagated by specific
+   * code (ex: user space queueing) so expect it to be null sometimes.
+   */
+  getCurrentContext(): ContextStore | null;
```

Comment:
We should also choose if we return a empty object or null when there are no active context. I would prefer to use null there since users would not trust it's content.

Reply:
I general I dislike `null` when we can avoid it, but I think you have a good point here - there is a semantic distinction between an empty object just because the context is empty and an empty object because there is no context available. I think that distinction is useful to preserve, so I like keeping it.

Reply:
I agree that `null` better clarifies the intent.

Reply:
I generally prefer smaller method names when possible. Any chance we could use something like `current()`? For a Context Manager, I would expect that to return the current context since that's what is managed by the class.

Reply:
@rochdev I don't really mind using `current()` but we should be consistent on this subject, do we avoid using get/set prefix or use them everywhere.
Seems that other PR's tend to go without prefix so i will update to match that.

Reply:
I agree we should be consistent. In general I tend to prefer omitting `get` since it's usually redundant. Verbs can be used for actions and nouns for getters. The only downside is for words that can be both, but then we can add `get` if the API would otherwise be unclear.